src/material/form-field/form-field.ts

```diff
@@ -507,7 +507,7 @@ export class MatFormField extends _MatFormFieldMixinBase
     if (this._control) {
       let ids: string[] = [];
 
-      if (this._control.userAriaDescribedBy) {
+      if (typeof this._control.userAriaDescribedBy === 'string') {
```

Comment:
Can you add a comment on why this assertion is needed? It's not clear to me in what cases `userAriaDescribedBy` becomes an object.

Reply:
My hypothesis would be that the lack of template type checking somewhere is letting someone bind an invalid value into something that then makes its way into `aria-describedby`. What we would typically want to do here is get to the root cause of the issue and potentially fix the offending app rather than accommodating an invalid value since the type system is supposed to provide this kind of protection.

Reply:
Yeah, that seems more likely to me. That's where I was heading. It seems wrong to change form-field to in favor of something we don't fully understand yet.

Reply:
Unless we can find the root cause of this type error, is the only other alternative to revert the entire PR?

Here's the PR that added this change
https://github.com/angular/components/pull/19587

Reply:
Who can look into this? I don't think I have the ability to help with that, unfortunately. To me it also sounds like someone explicitly passes an object to `aria-describedby` using an Angular input binding. It should be possible to filter out this app from the stack trace where the form-field threw an error. How many instances of these errors did you see?

Reply:
Just one. I'm also on vacation rn but I can track it down in my spare time. I'm not sure how long it'll take, though. Would we be ok with leaving this patch fix for now and adding a todo saying once this is tracked down we should remove this check?

Reply:
Yeah, surely works for me. I'll leave that to you. just wanted to make sure that this is temporary/fixed up in the future.

Reply:
I'll take the responsibility for this. You guys can call me out for it if this isn't resolved or at least updated within the next 2 weeks. I also created an issue to make sure we don't lose track of this https://github.com/angular/components/issues/20403

Reply:
More information: it's the Google Fonts app where this fails, and nowhere in the app do they set an `aria-describedby`, so it seems like it's something that's coming from us.